src/main/java/uk/gov/companieshouse/mapper/DissolutionDirectorResponseMapper.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.companieshouse.mapper;
+
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.model.db.dissolution.Dissolution;
+import uk.gov.companieshouse.model.dto.dissolution.DissolutionDirectorPatchResponse;
+import uk.gov.companieshouse.model.dto.dissolution.DissolutionLinks;
+
+@Service
+public class DissolutionDirectorResponseMapper {
+
+    public DissolutionDirectorPatchResponse mapToDissolutionDirectorPatchResponse(Dissolution dissolution) {
+        final DissolutionDirectorPatchResponse response = new DissolutionDirectorPatchResponse();
+
+        response.setLinks(generateLinks(
+                dissolution.getCompany().getNumber(),
+                dissolution.getData().getApplication().getReference()
+        ));
+
+        return response;
+    }
+
+    private DissolutionLinks generateLinks(String companyNumber, String reference) {
+        final DissolutionLinks links = new DissolutionLinks();
+
+        links.setSelf(String.format("/dissolution-request/%s", companyNumber));
```

Comment:
shouldn't this be app ref instead of company number?

Reply:
Aren't non-payment endpoints still using company number?

Reply:
oops - you are corrrect